modules/pricer/src/main/java/com/opengamma/strata/pricer/credit/cds/IsdaCdsProductPricer.java

```diff
@@ -0,0 +1,743 @@
+import com.opengamma.strata.product.credit.ResolvedCds;
+
+/**
+ * Pricer for single-name credit default swaps (CDS) based on ISDA standard model. 
+ * <p>
+ * The implementation is based on the ISDA model versions 1.8.2.
+ */
+public class IsdaCdsProductPricer {
+
+  /**
+   * Default implementation.
+   */
+  public static final IsdaCdsProductPricer DEFAULT = new IsdaCdsProductPricer(AccrualOnDefaultFormulae.ORIGINAL_ISDA);
+
+  /**
+   * The formula
+   */
+  private final AccrualOnDefaultFormulae formula;
+
+  /**
+   * The omega parameter.
+   */
+  private final double omega;
+
+  /**
+   * Constructor specifying the formula to use for the accrued on default calculation.  
+   * <p>
+   * Options are the formula given in the ISDA model (version 1.8.2 and lower); 
+   * the proposed fix by Markit (given as a comment in  version 1.8.2), or the mathematically correct formula. 
+   * <p>
+   * @param formula  the formula
+   */
+  public IsdaCdsProductPricer(AccrualOnDefaultFormulae formula) {
+    this.formula = ArgChecker.notNull(formula, "formula");
+    if (formula == AccrualOnDefaultFormulae.ORIGINAL_ISDA) {
+      omega = 1d / 730d;
+    } else {
+      omega = 0d;
+    }
+  }
+
+  //-------------------------------------------------------------------------
+  /**
+   * Calculates the present value of the CDS product.
+   * <p>
+   * The present value of the product is based on {@code referenceDate}.
+   * This is typically the valuation date, or cash settlement date if the product is associated with a {@code Trade}. 
+   * <p>
+   * The price type is clean or dirty. The accrued interest is computed based on the valuation date.
+   * 
+   * @param cds  the product
+   * @param ratesProvider  the rates provider
+   * @param referenceDate  the reference date
+   * @param priceType  the price type
+   * @param refData  the reference data
+   * @return the present value
+   */
+  public CurrencyAmount presentValue(
+      ResolvedCds cds,
+      CreditRatesProvider ratesProvider,
+      LocalDate referenceDate,
+      PriceType priceType,
+      ReferenceData refData) {
+
+    double price = price(cds, ratesProvider, referenceDate, priceType, refData);
+    return CurrencyAmount.of(cds.getCurrency(), cds.getBuySell().normalize(cds.getNotional()) * price);
+  }
+
+  /**
+   * Calculates the price of the CDS product, i.e., the present value per unit notional. 
+   * <p>
+   * This is coherent with {@link #presentValue(ResolvedCds, CreditRatesProvider, LocalDate, PriceType, ReferenceData)}.
+   * 
+   * @param cds  the product
+   * @param ratesProvider  the rates provider
+   * @param referenceDate  the reference date
+   * @param priceType  the price type
+   * @param refData  the reference data
+   * @return the price
+   */
+  public double price(
+      ResolvedCds cds,
+      CreditRatesProvider ratesProvider,
+      LocalDate referenceDate,
+      PriceType priceType,
+      ReferenceData refData) {
```

Comment:
It would be nice to avoid the need to pass in `ReferenceData`.
See `OvernightCompoundedRateComputation.calculatePublicationFromFixing` which might be a solution here.